File: app/src/main/java/com/jintao/vipmanager/database/bean/DbConvertGoodsInfo.java
package com.jintao.vipmanager.database.bean;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Property;
import org.greenrobot.greendao.annotation.Generated;

/**
 * Author: jintao
 * CreateDate: 2024/1/3 16:45
 * Description:兑换记录表
 */
@Entity(nameInDb = "convert_goods_info")
public class DbConvertGoodsInfo {

    @Id(autoincrement = true)
    private Long id;

    @Property(nameInDb = "uid")
    private int uid = 0;

    @Property(nameInDb = "goods_name")
    private String goodsName = "";

    @Property(nameInDb = "use_integral")
    private int useIntegral = 0;

    @Property(nameInDb = "desc")
    private String desc = "";

    @Property(nameInDb = "time")
    private String time = "";

    @Generated(hash = 120458840)
    public DbConvertGoodsInfo(Long id, int uid, String goodsName, int useIntegral,
            String desc, String time) {
        this.id = id;
        this.uid = uid;
        this.goodsName = goodsName;
        this.useIntegral = useIntegral;
        this.desc = desc;
        this.time = time;
    }

    @Generated(hash = 1817933744)
    public DbConvertGoodsInfo() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getUid() {
        return uid;
    }

    public void setUid(int uid) {
        this.uid = uid;
    }

    public String getGoodsName() {
        return goodsName;
    }

    public void setGoodsName(String goodsName) {
        this.goodsName = goodsName;
    }


    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public int getUseIntegral() {
        return useIntegral;
    }

    public void setUseIntegral(int useIntegral) {
        this.useIntegral = useIntegral;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }
}

File: app/src/main/java/com/jintao/vipmanager/database/bean/DbConvertGoodsInfo.kt
package com.jintao.vipmanager.database.bean

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Author: jintao
 * CreateDate: 2024/1/3 16:45
 * Description:兑换记录表
 */
@Entity(tableName = "exchange_goods_info")
class DbConvertGoodsInfo {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private var id:Long? = 0L

    @ColumnInfo(name = "uid")
    private var uid:Int = 0

    @ColumnInfo(name = "goods_name")
    private var goodsName:String? = ""

    @ColumnInfo(name = "use_integral")
    private var useIntegral:Int = 0

    @ColumnInfo(name = "desc")
    private var desc:String? = ""

    @ColumnInfo(name = "time")
    private var time:String? = ""

    fun getId():Long {
        return id!!
    }

    fun setId(id:Long) {
        this.id = id
    }

    fun getUid():Int {
        return uid
    }

    fun setUid(uid:Int) {
        this.uid = uid
    }

    fun getGoodsName():String {
        if(this.goodsName==null) {
            return ""
        }else {
            return this.goodsName!!
        }
    }

    fun setGoodsName(goodsName:String) {
        this.goodsName = goodsName;
    }

    fun getDesc():String {
        if(this.desc==null) {
            return ""
        }else {
            return this.desc!!
        }
    }

    fun setDesc(desc:String) {
        this.desc = desc
    }

    fun getUseIntegral():Int {
        return useIntegral
    }

    fun setUseIntegral(useIntegral:Int) {
        this.useIntegral = useIntegral;
    }

    fun getTime():String {
        if(this.time==null) {
            return ""
        }else {
            return this.time!!
        }
    }

    fun setTime(time:String) {
        this.time = time
    }
}
